### PR TITLE
fix(fmt): quick-fix create new block

### PR DESCRIPTION
**Pull request by Druue**
Something broke in the definition for these quickfixes in language-tools so this is an opportunistic attempt to both fix and move them to engines.

Reference: https://github.com/prisma/language-tools/blob/53e70077abbcf26c03c8aabb416521dba248804b/packages/language-server/src/lib/code-actions/index.ts#L137-L166

fixes prisma/team-orm#1144

### CODE REVIEW

File: prisma-fmt/src/code_actions/block.rs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    let span = Span {
+        start: model.ast_model().span().start,
+        end: model.ast_model().span().end + 1, // * otherwise it's still not outside the closing brace
+        file_id: model.ast_model().span().file_id,
+    };
+
+    let range = super::range_after_span(context.initiating_file_source(), span);
+
+    diagnostics.iter().for_each(|diag| {
+        push_missing_block(
```
Comment (SevInf) [inline prisma-fmt/src/code_actions/block.rs:30]: Nice thing we can do for mongo specifically: don't suggest to create model if we are inside of a composite type (composite types can not reference models).
Not necessary though, so feel free to ignore.

**Review by SevInf — APPROVED** [review]
Approved with one optional change suggestion

File: prisma-fmt/src/code_actions/block.rs (hunk 1 above)
Comment (janpio) [reply prisma-fmt/src/code_actions/block.rs:30]: (If we don't add now, turn into issue)

Comment (Druue) [reply prisma-fmt/src/code_actions/block.rs:30]: This is already only suggested from inside a model (& view).

```rs
/// code_actions.rs
// models AND views
    for model in validated_schema
        .db
        .walk_models_in_file(initiating_file_id)
        .chain(validated_schema.db.walk_views_in_file(initiating_file_id))
    {
        block::create_missing_block(&mut actions, &context, model);
```

I can add one for composite types to suggest further composite types. I don't think enums can hold _any_ references, if I'm not mistaken

Comment (SevInf) [reply prisma-fmt/src/code_actions/block.rs:30]: I had a false memory that `walk_models` also includes composites. It does not, you are right. 

>I can add one for composite types to suggest further composite types.

Let's do it. Would you prefer to do it now or later?

>I don't think enums can hold any references, if I'm not mistaken
Yes, I don't think there is anything more we can do for enums specifically

Comment (Druue) [reply prisma-fmt/src/code_actions/block.rs:30]: Actually, that comment is out of date, `walk_models` no longer includes views and we have a separate views walker now apparently

**Review by SevInf — APPROVED** [review]
Re-approving after latest changes, should be good to go now

**Pull request merged**